actions/create_event_handler.go

```diff
@@ -2,139 +2,76 @@ package actions
 
 import (
 	"context"
-	"encoding/json"
 	"fmt"
-	"strings"
+	"sync"
 
+	"github.com/castai/cluster-controller/castai"
 	"github.com/sirupsen/logrus"
 	v1 "k8s.io/api/core/v1"
-	"k8s.io/apimachinery/pkg/api/errors"
-	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	"k8s.io/apimachinery/pkg/types"
-	"k8s.io/apimachinery/pkg/util/strategicpatch"
 	"k8s.io/client-go/kubernetes"
-
-	"github.com/castai/cluster-controller/castai"
+	typedv1core "k8s.io/client-go/kubernetes/typed/core/v1"
+	"k8s.io/client-go/tools/record"
```

Comment:
From which k8s version is this package available?

Reply:
record package had been there for a while, EventRecorder interface is already 5 years old. New events API was promoted to GA in 1.19 https://github.com/kubernetes/enhancements/tree/master/keps/sig-instrumentation/383-new-event-api-ga-graduation#implementation-history